guides/source/action_view_helpers.md

```diff
@@ -5,668 +5,1054 @@ Action View Helpers
+
+See the [API
+Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/JavaScriptHelper.html#method-i-escape_javascript)
+for more information.
 
-Formats the bytes in size into a more understandable representation; useful for reporting file sizes to users.
+### javascript_tag
+
+Returns a JavaScript tag wrapping the provided code. You can pass a hash of
+options to control the behavior of the `<script>` tag.
 
 ```ruby
-number_to_human_size(1234)    # => 1.21 KB
-number_to_human_size(1234567) # => 1.18 MB
+javascript_tag("alert('All is good')", type: "application/javascript")
 ```
 
-See the [API Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_to_human_size) for more information.
+```html
+<script type="application/javascript">
+//<![CDATA[
+alert('All is good')
+//]]>
+</script>
+```
 
-#### number_to_percentage
+Instead of passing the content as an argument, you can also use a block.
 
-Formats a number as a percentage string.
+```html+erb
+<%= javascript_tag type: "application/javascript" do %>
+  alert("Welcome to my app!")
+<% end %>
+```
+
+See the [API
+Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/JavaScriptHelper.html#method-i-javascript_tag)
+for more information.
+
+Alternative Tags
+----------------------------------------------
+
+A set of methods to generate HTML tags programmatically.
+
+### tag
+
+Generates a standalone HTML tag with the given `name` and `options`.
+
+Every tag can be built with:
 
 ```ruby
-number_to_percentage(100, precision: 0) # => 100%
+tag.some_tag_name(optional content, options)
 ```
 
-See the [API Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_to_percentage) for more information.
+where tag name can be e.g. `br`, `div`, `section`, `article`, or any tag really.
 
-#### number_to_phone
-
-Formats a number into a phone number (US by default).
+For example, here are some common uses:
 
 ```ruby
-number_to_phone(1235551234) # => 123-555-1234
+tag.h1 "All titles fit to print"
+# => <h1>All titles fit to print</h1>
+
+tag.div "Hello, world!"
+# => <div>Hello, world!</div>
 ```
 
-See the [API Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_to_phone) for more information.
+#### Options
 
-#### number_with_delimiter
+You can use symboled key options to add attributes to the generated tag.
 
-Formats a number with grouped thousands using a delimiter.
+```ruby
+tag.section class: %w( kitties puppies )
+# => <section class="kitties puppies"></section>
+```
+
+In addition, HTML `data-*` attributes can be passed to the `tag` helper as a key value pair, where the key is `data` and the value is a hash containing sub-attributes. The sub-attributes are then converted to `data-*` attributes that are dasherized in order to work well with JavaScript.
```

Comment:
```suggestion
In addition, HTML `data-*` attributes can be passed to the `tag` helper using the `data` option, with a hash containing key-value pairs of sub-attributes. The sub-attributes are then converted to `data-*` attributes that are dasherized in order to work well with JavaScript.
```

Note: Make sure to format the column length on this last block to match the rest of the guide.